### PR TITLE
Use new-style travis config

### PR DESCRIPTION
**Pull request by chillu**
- Fixes build failures with missing versioned extension
- Introduces multi database builds

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-        </testsuite>
-    </testsuites>
+<phpunit bootstrap="framework/tests/bootstrap.php" colors="true">
+
+    <testsuite name="Default">
+        <directory>tests</directory>
+    </testsuite>
+
+    <listeners>
+        <listener class="SilverStripe\Dev\TestListener" />
```
Comment (tractorcow) [inline phpunit.xml]: This class was deleted... not sure how this is passing.

File: .travis.yml [hunk 2]
```diff
@@ -1,24 +1,40 @@
 language: php
 
+sudo: false
+
 env:
   global:
-    - DB=MYSQL CORE_RELEASE=master
+    - COMPOSER_ROOT_VERSION=1.0.x-dev
+    - CORE_RELEASE=master
```
Comment (tractorcow) [inline .travis.yml]: I think CORE_RELEASE can be retired now.

File: composer.json [hunk 3]
```diff
@@ -21,12 +21,15 @@
...
     "psr-4": {
-      "SilverStripe\\GraphQL\\Tests\\": "tests/"
+      "SilverStripe\\GraphQL\\Tests\\": "tests/php/"
     }
   },
   "config": {
     "process-timeout": 600
   },
+  "scripts": {
+    "lint": "phpcs --standard=tests/phpcs/ruleset.xml src/ tests/"
```
Comment (tractorcow) [inline composer.json]: I've found that if you rename this file to `phpcs.xml.dist` and put it in the root it's found automatically and doesn't need to be specified.

File: phpunit.xml (hunk 1 above)
Comment (tractorcow) [inline phpunit.xml]: Maybe the file needs to be named `phpunit.xml.dist`?

File: composer.json (hunk 3 above)
Comment (tractorcow) [inline composer.json]: See CMS for example.

Comment (sminnee) [inline composer.json]: +1 to using `phpcs.xml.dist` as this will mean that IDEs pick it up too

File: .travis.yml [hunk 4]
```diff
@@ -1,24 +1,40 @@
...
     - php: 7.0
+      env: DB=MYSQL PHPUNIT_TEST=1
     - php: 7.1
+      env: DB=MYSQL PDO=1 PHPUNIT_COVERAGE_TEST=1
 
 before_script:
-    - git clone git://github.com/silverstripe/silverstripe-travis-support.git ~/travis-support
-    - php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss
-    - cd ~/builds/ss
+  - printf "\n" | pecl install imagick
```
Comment (sminnee) [inline .travis.yml]: Is this needed for graphql testing?

File: .travis.yml [hunk 5]
```diff
@@ -1,24 +1,40 @@
...
+    - COMPOSER_ROOT_VERSION=1.0.x-dev
+    - CORE_RELEASE=master
 
 matrix:
+  fast_finish: true
   include:
     - php: 5.6
+      env: DB=MYSQL PHPUNIT_TEST=1 PHPCS_TEST=1
+    - php: 5.6
+      env: DB=SQLITE PHPUNIT_TEST=1
```
Comment (sminnee) [inline .travis.yml]: Should we ditch SQLITE builds in favour of PGSQL ones?

File: .travis.yml (hunk 2 above)
Comment (sminnee) [inline .travis.yml]: Yep

File: composer.json (hunk 3 above)
Comment (chillu) [inline composer.json]: Right, I was following the framework module "best practice" heh - so much fragmentation in build config!

File: .travis.yml (hunk 4 above)
Comment (chillu) [inline .travis.yml]: Nope, I've removed it

File: .travis.yml (hunk 5 above)
Comment (chillu) [inline .travis.yml]: As discussed on Slack, we agree that SQLite and PG can be run through the nightly installer build only

File: .travis.yml (hunk 2 above)
Comment (chillu) [inline .travis.yml]: Done

File: phpunit.xml (hunk 1 above)
Comment (chillu) [inline phpunit.xml]: Done

**Comment by sminnee** [comment]
Automated codecov/project is benign AFAIK — I'm happy for this to be merged.

**Pull request merged**